ouroboros-consensus/src/Ouroboros/Consensus/Block/Forging.hs

```diff
@@ -94,34 +135,87 @@ data BlockForging m blk = BlockForging {
         -> m blk
     }
 
-getLeaderProof ::
+data ShouldForge blk =
+    -- | We are a leader in this slot, but we should not forge because
```

Comment:
```suggestion
    -- | We are a leader in this slot, but we should not forge because
```
We are not necessarily a leader, we haven't even checked yet.